### PR TITLE
rstream [RFC]: Set INLINE data for iWARP

### PR DESCRIPTION
**Pull request by tatyana-en**
Set rsocket SQ INLINE to RS_MSG_SIZE for iWARP.

### CODE REVIEW

**Comment by rleon** [comment]
@tatyana-en , rstream as an example application, why is this change marked as RFC?

**Comment by tatyana-en** [comment]
It is marked as RFC to get comments, since I didn't discuss it with an rstream maintainer.

**Comment by rleon** [comment]
> It is marked as RFC to get comments, since I didn't discuss it with an rstream maintainer.

Are you referring to Sean? @shefty

File: librdmacm/examples/rstream.c [hunk 1]
```diff
@@ -304,7 +304,7 @@ static void set_options(int fd)
 			rs_setsockopt(fd, SOL_RDMA, RDMA_INLINE, &inline_size,
 				      sizeof inline_size);
 		} else if (optimization == opt_bandwidth) {
-			val = 0;
+			val = 4;
```
Comment (shefty) [inline librdmacm/examples/rstream.c:307]: I'm soooo out of touch with this code...  One could argue the problem is actually in rs_create_ep() and not rstream.  The spot in rs_create_ep() where it does this:
```
    if (rs->cm_id->verbs->device->transport_type == IBV_TRANSPORT_IWARP)
        rs->opts |= RS_OPT_MSG_SEND;
```
adjusts the rsocket options, which causes the check below (also in rs_create_ep()) to fail:
```
    if ((rs->opts & RS_OPT_MSG_SEND) && (rs->sq_inline < RS_MSG_SIZE))
        return ERR(ENOTSUP);
```
Technically, there's nothing a use did wrong here.  I think a better fix is to handle this internally.  Something like:
```
    /* See comment for RS_OPT_MSG_SEND */
    if (rs->cm_id->verbs->device->transport_type == IBV_TRANSPORT_IWARP) {
        rs->opts |= RS_OPT_MSG_SEND;
        if (rs->sq_inline < RS_MSG_SIZE)
            rs->sq_inline = RS_MSG_SIZE;
    }
```

Comment (tatyana-en) [reply librdmacm/examples/rstream.c]: Thanks for the feedback!

**Comment by shefty** [comment]
I meant to the pull request itself.  That is, where is says "rstream [RFC]: Set INLINE data for iWARP".  It doesn't really matter, since it doesn't end up in the git logs, but can be helpful for whoever is managing the pull requests.

**Comment by tatyana-en** [comment]
> I meant to the pull request itself. That is, where is says "rstream [RFC]: Set INLINE data for iWARP". It doesn't really matter, since it doesn't end up in the git logs, but can be helpful for whoever is managing the pull requests.

Unfortunately, I don't know how to do this without closing this PR and opening a new one with a different name.

**Pull request merged**